### PR TITLE
Pseudocode and explanation added

### PR DESCRIPTION
**Pull request by ghost**
If there's anything else worth adding to the explanation let me know.

### CODE REVIEW

File: Assn3_Pseudo.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+int main()
+   var plaintext
+   get(plaintext)                                        // get text to be hashed
+   var[] blocks = getBlocks(plaintext.AlNumOnly)         // separate into blocks
+   var[] intBlocks = operation(blocks, AlNumOnly.length) // ?
```
Comment (andres-imperial) [inline Assn3_Pseudo.txt]: perform hashing operation

File: Assn3_Pseudo.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+int main()
+   var plaintext
+   get(plaintext)                                        // get text to be hashed
+   var[] blocks = getBlocks(plaintext.AlNumOnly)         // separate into blocks
+   var[] intBlocks = operation(blocks, AlNumOnly.length) // ?
+   print("Hash: " + fixLength(intBlocks)                 // print hashed value
+
+// create blocks from a shifted bitwise OR of each 4-character chunk
```
Comment (andres-imperial) [inline Assn3_Pseudo.txt]: create 4 character blocks of 32 bits, from plaintext

File: Assn3_Pseudo.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      chunk = 0
+      for (j = 0 to 4)                           // for each character in chunk
+         chunk = chunk << 8                      // shift chunk left 8 bits
+         if (j + i < plaintext.length)
+            var temp = plaintext(j+i)
+            chunk = chunk | temp                 // bitwise OR with current char
+      blocks.push(chunk)
+   return blocks
+
+// bitwise OR a left and right shift of each block to the power of the total length
```
Comment (andres-imperial) [inline Assn3_Pseudo.txt]: for each block raise it to the power of the plaintext's message length, then perform modulus 32 on resulting value to obtain value for rotational shift.

File: Assn3_Explanation.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Efficient: Out algorithm mostly consists of bitwise ORs and shifts, which are not very intensive operations. The most expensive task is the pow() in the operation() function, and it only happens once for each block.
```
Comment (StephGonz) [inline Assn3_Explanation.txt]: Did you mean "Our" instead of "Out" here?

File: Assn3_Explanation.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Efficient: Out algorithm mostly consists of bitwise ORs and shifts, which are not very intensive operations. The most expensive task is the pow() in the operation() function, and it only happens once for each block.
+
+Deterministic: There are no random operations anywhere in our algorithm, so the same string will have to produce the same hash value each time.
+
+Pre-Image Resistant: Each block of 4 characters is combined with a bitwise OR and the blocks are added together at the end, so any change to an individual character will change the entire block, which will then significantly change the final hash.
+
+Collision Resistant: We have a fixed output of 128 bits that comes from each character's binary code being merged together, so the likelihood of another string producing the same hash is negligible.
```
Comment (StephGonz) [inline Assn3_Explanation.txt]: I would consider rewording this. Technically we don't have a completely fixed output of 128 bits. It is more of a fixed **max** output of 128 bits. Some hashes could technically be smaller but they won't be larger than 128.

Comment (StephGonz) [reply Assn3_Explanation.txt]: Honestly, I would just add the word 'max' between 'fixed' and 'output' and leave the rest alone.

**Pull request merged**